packages/kernel/src/kernel.ts

```diff
@@ -534,6 +537,25 @@ export abstract class BaseKernel implements IKernel {
     this._sendMessage(message);
   }
 
+  /**
+   * Handle an input_request message
+   *
+   * @param msg The parent message.
+   */
+  private async _inputRequest(msg: KernelMessage.IMessage): Promise<void> {
+    const inputMsg = msg as KernelMessage.IInputRequestMsg;
+    const content = await this.inputRequest(inputMsg.content);
```

Comment:
This is not good. Subclasses should not implement an abstract `inputRequest` method.

This is a kernel-to-UI message, just like messages on the `iopub`. So this should look more like those messages.
Something like
```typescript
    const message = KernelMessage.createMessage<KernelMessage.IInputRequestMsg>({
      channel: 'stdin',
      msgType: 'input_request',
      // TODO: better handle this
      session: this._parentHeader?.session ?? '',
      parentHeader: this._parentHeader,
      content
    });
    this._sendMessage(message);
```